packages/core/src/tools/utils.ts

```diff
@@ -304,3 +307,52 @@ export function safeTruncate(candidate: string, length: number) {
   }
   return candidate.slice(0, length)
 }
+
+export function addGlobalEventListeners(
+  events: DOM_EVENT[],
+  listener: (event: Event) => void,
+  { once, capture }: { once?: boolean; capture?: boolean } = {}
+): { stop(): void } {
+  const wrapedListener = monitor(
+    once
+      ? (event: Event) => {
+          stop()
+          listener(event)
+        }
+      : listener
+  )
+
+  const options = supportPassiveEventListeners() ? { capture, passive: true } : capture
+  events.forEach((event) => addEventListener(event, wrapedListener, options))
+  const stop = () => events.forEach((event) => removeEventListener(event, wrapedListener, options))
+
+  return {
+    stop,
+  }
+}
+
+let supportPassiveEventListenersCache: boolean | undefined
+// Inspired by
+// https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#Safely_detecting_option_support
+function supportPassiveEventListeners() {
+  if (supportPassiveEventListenersCache === undefined) {
+    supportPassiveEventListenersCache = false
+
+    try {
+      const options: AddEventListenerOptions = {
+        get passive() {
+          supportPassiveEventListenersCache = true
+          return false
+        },
+      }
+
+      // tslint:disable-next-line: no-null-keyword
+      addEventListener('test' as any, null as any, options)
+      // tslint:disable-next-line: no-null-keyword
+      removeEventListener('test' as any, null as any, options)
```

Comment:
what about using `noop` instead of `null`?

Reply:
Ok, but let's see first if we should remove this check altogether